### PR TITLE
Feature/edit routing

### PR DESCRIPTION
**Pull request by DoctorFreed**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
+import Edit from "./Edit";
 
 const App = () => {
   const [allTask, changeAllTask] = useState([]);
-  const [textInput, changeTextInput] = useState("");
   const [showAlert, changeShowAlert] = useState(false);
   const [alertBody, changeAlertBody] = useState({});
+  const [taskEdit, setTaskEdit] = useState();
+
+  let history = useHistory();
```
Comment (usatovaDasha) [inline src/App.js:15]: why let?

File: src/Edit.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          type="text"
+          value={textInput}
+          placeholder="Input your task"
+          onChange={(e) => setTextInput(e.target.value)}
+        />
+        <Button
+          variant="primary"
+          onClick={() => {
+            props.editTask({
+              _id: props.taskEdit._id,
```
Comment (usatovaDasha) [inline src/Edit.js:27]: destructurization props

File: src/Home.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { useState } from "react";
+import { Button, Form } from "react-bootstrap";
+import Task from "./Task";
+
+const Home = (props) => {
```
Comment (usatovaDasha) [inline src/Home.js:6]: destructurization props

File: src/Home.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  const [textInput, changeTextInput] = useState("");
+
+  return (
+    <div className="App">
+      <div className="c-title-todo">
+        <h1 className="text-title-todo">to-do list</h1>
+      </div>
+      <div className="c-input-group">
+        <Form.Control
```
Comment (usatovaDasha) [inline src/Home.js:15]: why it is not new component?

File: src/Task.js [hunk 5]
```diff
@@ -1,48 +1,19 @@
-import React, { useEffect, useState } from "react";
+import React, { useState } from "react";
 import { Form, Card } from "react-bootstrap";
 import "./Task.scss";
 import editImg from "./img/edit.svg";
 import deleteImg from "./img/delete.svg";
-import doneImg from "./img/done.svg";
-import closeImg from "./img/close.svg";
 
 const Task = (props) => {
```
Comment (usatovaDasha) [inline src/Task.js:7]: destructurization props

File: src/App.js [hunk 6]
```diff
@@ -93,45 +106,32 @@ const App = () => {
...
-              delete={deleteTask}
-              edit={editTask}
-              {...item}
-            />
-          );
-        })}
+      <Switch>
+        <Route path="/home">
+          <Home
+            deleteTask={deleteTask}
```
Comment (usatovaDasha) [inline src/App.js:123]: why are these functions not included in the Task component?

File: src/App.js (hunk 1 above)
Comment (DoctorFreed) [reply src/App.js:15]: yep true, fixed that

File: src/Edit.js (hunk 2 above)
Comment (DoctorFreed) [reply src/Edit.js:27]: done

File: src/Home.js (hunk 3 above)
Comment (DoctorFreed) [reply src/Home.js:6]: done

File: src/Task.js (hunk 5 above)
Comment (DoctorFreed) [reply src/Task.js:7]: done

File: src/App.js (hunk 6 above)
Comment (DoctorFreed) [reply src/App.js:123]: all interaction with api is in app.js, home.js transmits that function to task.js

File: src/Home.js (hunk 4 above)
Comment (DoctorFreed) [reply src/Home.js:15]: thought about it, okay i create a new component

File: src/Edit.js [hunk 7]
```diff
@@ -23,10 +23,10 @@ const Edit = (props) => {
         <Button
           variant="primary"
           onClick={() => {
-            props.editTask({
-              _id: props.taskEdit._id,
+            editTask({
+              _id: taskEdit._id,
```
Comment (usatovaDasha) [inline src/Edit.js:27]: destructurization taskEdit

File: src/Task.js [hunk 8]
```diff
@@ -4,20 +4,20 @@ import "./Task.scss";
 import editImg from "./img/edit.svg";
 import deleteImg from "./img/delete.svg";
 
-const Task = (props) => {
-  const [checkedTask, changeCheckedTask] = useState(props.isCheck);
+const Task = ({ deleteItem, edit, goEdit, task }) => {
+  const [checkedTask, changeCheckedTask] = useState(task.isCheck);
```
Comment (usatovaDasha) [inline src/Task.js:8]: destructurization task

File: src/App.js (hunk 6 above)
Comment (usatovaDasha) [reply src/App.js:123]: just pass the state changes to the global. And that's it, and the functions can lie inside the components for which it is intended and where their elements are located

File: src/Edit.js (hunk 7 above)
Comment (DoctorFreed) [reply src/Edit.js:27]: done

File: src/Task.js (hunk 8 above)
Comment (DoctorFreed) [reply src/Task.js:8]: done

File: src/App.js (hunk 6 above)
Comment (DoctorFreed) [reply src/App.js:123]: i dont quite understand how it can be done, "global" where is it? in file index.js?

File: src/App.js [hunk 9]
```diff
@@ -128,7 +128,7 @@ const App = () => {
           />
         </Route>
         <Route path="/edit">
-          <Edit taskEdit={taskEdit} editTask={editTask} />
+          <Edit _id={taskEdit._id} text={taskEdit.text} isCheck={taskEdit.isCheck} editTask={editTask} />
```
Comment (usatovaDasha) [inline src/App.js:131]: previously it was written correctly

File: src/Home.js [hunk 10]
```diff
@@ -18,7 +18,9 @@ const Home = ({ deleteTask, editTask, allTask, addTask, goEdit }) => {
               deleteItem={deleteTask}
               edit={editTask}
               goEdit={goEdit}
-              task={item}
+              _id={item._id}
```
Comment (usatovaDasha) [inline src/Home.js:21]: previously it was written correctly

File: src/Task.js [hunk 11]
```diff
@@ -35,13 +35,13 @@ const Task = ({ deleteItem, edit, goEdit, task }) => {
           <Card.Text
             onDoubleClick={() =>
               goEdit({
-                _id: task._id,
-                text: task.text,
-                isCheck: task.isCheck,
+                _id: _id,
```
Comment (usatovaDasha) [inline src/Task.js:38]: { _id, text, isCheck }

**Pull request merged**